### DISCUSSION TITLE
Support for code splitting

### DISCUSSION THREAD

**Discussion by kbrooks** [pmndrs/zustand#2229] [Ideas]
Hi, I work on a large website which uses Redux and we have logic to dynamically add reducers when a new bundle is loaded. This makes it possible to do code splitting on our Redux store, since reducers can be registered into the store when the code that requires them is loaded.

https://redux.js.org/usage/code-splitting

Is it possible to do the same with zustand?

**Comment by dbritto-dev** [comment]
@kbrooks I guess it could be possible using a middleware on top of redux middleware

**Reply by dbritto-dev** [reply]
I'll try to came up with something I've been thinking on this but for slices or scoped slices

**Comment by dai-shi** [comment]
Or, if you are not focusing on reducers, https://docs.pmnd.rs/zustand/guides/practice-with-no-store-actions should be easier.

**Reply by kbrooks** [reply]
This should work. It would be nice if state could also be initialized but that is not a requirement. 

For some reason this page does not come up when you search "code splitting" in the Zustand docs

**Reply by dai-shi** [reply]
The search feature might not be working as expected. cc @CodyJasonBennett
